iceoryx2-bb/testing/src/memory.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use std::{
+    alloc::{alloc, dealloc, Layout},
+    ops::{Deref, DerefMut},
+};
+
+/// Allocates uninitialized memory on the heap. When it goes out of scope the memory is released.
+/// The user has to ensure that the memory is initialized.
+pub struct RawMemory<T> {
+    memory: *mut T,
+}
+
+impl<T> Drop for RawMemory<T> {
+    fn drop(&mut self) {
+        unsafe { dealloc(self.memory.cast(), Layout::new::<T>()) }
+    }
+}
+
+impl<T> Deref for RawMemory<T> {
+    type Target = T;
+    fn deref(&self) -> &Self::Target {
+        unsafe { &*self.memory }
+    }
```

Comment:
This is unsound. The user can access `T` which might contain a bit patter which is not representative for a `T`.

It's convenient but something like `unsafe fn assume_init() -> &Self::Target` would be more safe.

Reply:
I thought this would be valid since it is only for testing but I guess you are right